include/ranking_response.h

```diff
@@ -16,41 +16,28 @@
 namespace reinforcement_learning {
   class api_status;
 
-  /**
-   * @brief Holds (action, probability) pairs, POD used for extern "C"
-   */
-  struct action_prob_d {
-    //! action id
-    size_t action_id;
-    //! probability associated with the action id
-    float probability;
-  };
-
-  /**
-   * @brief Holds (action, probability) pairs.
-   */
-  struct action_prob : public action_prob_d {
-    inline action_prob(size_t action_id, float probability) {
-      this->action_id = action_id;
-      this->probability = probability;
-    };
-  };
-
   /**
    * @brief choose_rank() returns the action choice using ranking_response.
    * ranking_response contains all the actions and distribution from with the action was sampled.  It also contains the chosen action id and
    * the unique event_id representing the choice.  This unique event_id must be used to report back outcomes against this choice for the online trainer.
    */
   class ranking_response {
+  private:
+	using coll_t = std::vector<action_prob>;
```

Comment:
looks like it is cleaner to have dependency directly from slot_ranking::iterator instead of redefining new coll_t type (which is the same as slot_ranking::coll_t, but it is not enforced anyhow)